src/index.js

```diff
@@ -92,16 +92,22 @@ const addChecktext = () => {
 
 const deleteComplete = () => {
   const clearComplete = document.querySelector('.clear-complete');
+  let resArr;
   if (clearComplete && todo.check) {
     clearComplete.addEventListener('click', () => {
-      todo.check.sort();
-      for (let k = todo.check.length - 1; k >= 0; k -= 1) {
-        todo.taskArr.splice(todo.check[k], 1);
-      }
+      todo.check.sort((a, b) => b - a);
+      resArr = todo.taskArr.filter((task) => {
+        if (task.completed !== 'false') {
+          return false;
+        }
+        return task;
+      });
+      todo.taskArr = resArr;
```

Comment:
[OPTIONAL] I think could have just made it simpler by doing something like this:

```
todo.taskArr.filter((task) => task.completed === 'true'))
```